src/session_impl.cpp

```diff
@@ -1887,6 +1864,50 @@ namespace {
 		}
 	}
 
+	void session_impl::remove_listen_sockets(std::vector<listen_endpoint_t>& eps)
+	{
```

Comment:
it would be nice if this could be a free function, and unit testable. but I guess the access to ``m_dht`` makes that a bit tricky